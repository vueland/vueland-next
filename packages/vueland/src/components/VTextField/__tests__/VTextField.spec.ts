import { mount } from '@vue/test-utils'
import { VTextField } from '../index'
import { ref } from 'vue'
import 'regenerator-runtime'

describe('VTextField', () => {
  const val = ref('')
  const onUpdate = (v) => (val.value = v)
  const mountFunction = (options = {}) => mount(VTextField as any, options)

  beforeEach(() => {
    val.value = ''
  })

  it('should mount component and match snapshot', () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should test non disabled state', async () => {
    const wrapper = mountFunction({
      props: {
        modelValue: val.value,
        label: 'email',
        rules: [(val) => !!val || 'required input'],
        color: 'orange darken-1',
        textColor: '#fa5a5a',
        ['onUpdate:modelValue']: onUpdate,
      },
      global: {
        provide: {
          form: null,
        },
      },
    })

    await wrapper.find('input').trigger('focus')
    await wrapper.find('input').trigger('blur')

    expect(wrapper.find('.v-input').attributes('class')).toContain(
      'v-input--not-valid'
    )
    expect(wrapper.html()).toMatchSnapshot()

    val.value = 'sss'

    await wrapper.setProps({modelValue: val.value})

    expect(wrapper.find('.v-input--not-valid').exists()).toBeFalsy()

    expect(wrapper.find('label').text()).toContain('email')
    expect(wrapper.find('.v-input').attributes('class')).toContain(
      'orange--text text--darken-1'
    )
    expect(wrapper.find('input').attributes('style')).toContain(
      'color: rgb(250, 90, 90);'
    )
  })

  it('should test disabled state', async () => {
    const wrapper = mountFunction({
      props: {
        label: 'email',
        rules: [(val) => !!val || 'required input'],
        color: 'orange darken-1',
        textColor: '#fa5a5a',
        disabled: true,
      },
      global: {
        provide: {
          form: null,
        },
      },
    })

    expect(wrapper.find('input').attributes('style')).toBeFalsy()
    expect(wrapper.find('input[disabled]').exists()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.find('input').trigger('input')

    expect(wrapper.find('.v-input--focused').exists()).toBeFalsy()
  })
})
