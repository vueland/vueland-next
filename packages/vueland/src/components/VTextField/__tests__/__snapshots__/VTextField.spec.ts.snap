// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VTextField should mount component and match snapshot 1`] = `
<div class="v-input">
  <div class="v-input__label">
    <label class="v-label v-label--on-input">
    </label>
  </div>
  <div class="v-input__text-field">
    <div class="v-text-field">
      <input class="v-text-field__input"
             type="text"
      >
    </div>
  </div>
  <div class="v-input__hints">
    <transition-stub name="fade"
                     mode
                     appear="false"
                     persisted="false"
                     css="true"
    >
    </transition-stub>
  </div>
</div>
`;

exports[`VTextField should test disabled state 1`] = `
<div class="v-input v-input--disabled">
  <div class="v-input__label">
    <label class="v-label v-label--disabled v-label--on-input">
      email
    </label>
  </div>
  <div class="v-input__text-field">
    <div class="v-text-field v-text-field--disabled">
      <input class="v-text-field__input"
             label="email"
             rules="function (val) {
                  return !!val || 'required input';
                }"
             color="orange darken-1"
             textcolor="#fa5a5a"
             disabled
             type="text"
      >
    </div>
  </div>
  <div class="v-input__hints">
    <transition-stub name="fade"
                     mode
                     appear="false"
                     persisted="false"
                     css="true"
    >
    </transition-stub>
  </div>
</div>
`;

exports[`VTextField should test non disabled state 1`] = `
<div class="v-input v-input--not-valid error--text">
  <div class="v-input__label">
    <label class="v-label v-label--on-input">
      email
    </label>
  </div>
  <div class="v-input__text-field">
    <div class="v-text-field">
      <input class="v-text-field__input"
             style="color: rgb(250, 90, 90);"
             modelvalue
             label="email"
             rules="function (val) {
                  return !!val || 'required input';
                }"
             color="orange darken-1"
             textcolor="#fa5a5a"
             type="text"
      >
    </div>
  </div>
  <div class="v-input__hints">
    <transition-stub name="fade"
                     mode
                     appear="false"
                     persisted="false"
                     css="true"
    >
      <span class="v-input__hints-message">
        required input
      </span>
    </transition-stub>
  </div>
</div>
`;
